### PR TITLE
replacing mgo refernces with mongo-go-driver

### PR DESCRIPTION
**Pull request by knainwal**
## Description

a3s code currently uses direct mgo references in many places. Replacing those with mongo-go-driver equivalents in order to prepare for MongoDB upgrade.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: pkgs/api/Makefile [hunk 1]
```diff
@@ -8,9 +8,9 @@ all: format codegen lint test
 
 .PHONY:codegen
 codegen:
-	elegen folder -d specs -o codegen -g openapi3 || exit 1
+	/Users/knainwal/apomux/workspace/code/go/src/go.aporeto.io/elemental/cmd/elegen/elegen folder -d specs -o codegen -g openapi3 || exit 1
```
Comment (sawyerawesome) [inline pkgs/api/Makefile:11]: these shouldn't be committed

File: pkgs/api/Makefile [hunk 2]
```diff
@@ -8,9 +8,9 @@ all: format codegen lint test
...
 	rm -rf openapi3
-	elegen folder -d specs -o codegen || exit 1
+	/Users/knainwal/apomux/workspace/code/go/src/go.aporeto.io/elemental/cmd/elegen/elegen folder -d specs -o codegen || exit 1
```
Comment (sawyerawesome) [inline pkgs/api/Makefile:13]: and here

**Review by sawyerawesome — COMMENTED** [review]
added comments

File: pkgs/api/Makefile (hunk 1 above)
Comment (knainwal) [reply pkgs/api/Makefile:11]: Oops. Thanks for spotting this Sawyer. Accidentally left this over.

File: pkgs/api/Makefile (hunk 2 above)
Comment (knainwal) [reply pkgs/api/Makefile:13]: Fixing now.

**Review by sawyerawesome — APPROVED** [review]
LGTM

**Review by sawyerawesome — APPROVED** [review]
lgtm

**Pull request merged**